docs/intro/tutorial04.txt

```diff
@@ -20,16 +20,16 @@ tutorial, so that the template contains an HTML ``<form>`` element:
 .. code-block:: html+django
     :caption: polls/templates/polls/detail.html
 
-    <h1>{{ question.question_text }}</h1>
-
-    {% if error_message %}<p><strong>{{ error_message }}</strong></p>{% endif %}
-
     <form action="{% url 'polls:vote' question.id %}" method="post">
     {% csrf_token %}
-    {% for choice in question.choice_set.all %}
-        <input type="radio" name="choice" id="choice{{ forloop.counter }}" value="{{ choice.id }}">
-        <label for="choice{{ forloop.counter }}">{{ choice.choice_text }}</label><br>
-    {% endfor %}
+    <fieldset>
+        <legend><h1>{{ question.question_text }}</h1></legend>
```

Comment:
> But the `<h1>` implies that this is the heading for the complete page while the `<legend>` implies that this is only the heading for this fieldset.

In this case it's exactly the same, because the detail page contains only a single radio. IMO it's fine:

![image](https://user-images.githubusercontent.com/2865885/120777076-35159200-c525-11eb-8b9c-22299e705062.png)

Reply:
It does feel strange, but in this case it is indeed the case that this bit of text is both the page heading, and the `fieldset` ’s `legend`.

This is inspired by the example of the GOV.UK design system: https://design-system.service.gov.uk/components/radios/, and I tested this in VoiceOver to confirm this was announced as expected for VoiceOver users (haven’t tested with other screen readers but I suppose the GDS have)

The alternative would be to only have the `legend` and no `h1`, which I think would be acceptable as well, but for screen reader users who want to navigate by heading it’s definitely nicer that there is an `h1`.